coalib/results/TextRange.py

```diff
@@ -89,7 +89,20 @@ def end(self):
         return self._end
 
     def overlaps(self, other):
-        return self.start <= other.end and self.end >= other.start
+        return (self.is_behind(self.start, other.end) and
+                self.is_behind(other.start, self.end))
+
+    @staticmethod
+    def is_behind(first_position, second_position):
+        if first_position.line is None or second_position.line is None:
+            return True
+        if first_position.line == second_position.line:
+            return (True
+                    if None in (first_position.column, second_position.column)
+                    else
+                    first_position.column <= second_position.column)
```

Comment:
maybe for the sake of readability, use normal if's again, not sure any more... up to you :)